reconcile/templating/lib/merge_request_manager.py

```diff
@@ -211,8 +137,9 @@ def create_tr_merge_request(self, output: list[TemplateOutput]) -> None:
         collection_hash = collection_hashes.pop()
 
         """Create a new MR with the rendered template."""
-        if mr := self._merge_request_already_exists(collection):
-            if mr.template_info.collection_hash == collection_hash:
+        if mr := self._merge_request_already_exists({"collection": collection}):
+            assert isinstance(mr.mr_info, TemplateInfo)
```

Comment:
I also don't like those type cast asserts :(

Reply:
will look at it